api/client/contextdialer.go

```diff
@@ -34,23 +35,23 @@ type ContextDialer interface {
 	DialContext(ctx context.Context, network, addr string) (net.Conn, error)
 }
 
-// ContextDialerFunc is a function wrapper that implements the ContextDialer interface
+// ContextDialerFunc is a function wrapper that implements the ContextDialer interface.
 type ContextDialerFunc func(ctx context.Context, network, addr string) (net.Conn, error)
 
 // DialContext is a function that dials to the specified address
 func (f ContextDialerFunc) DialContext(ctx context.Context, network, addr string) (net.Conn, error) {
 	return f(ctx, network, addr)
 }
 
-// NewDialer makes a new dialer.
+// NewDialer makes a new dialer to connect directly to an Auth server.
 func NewDialer(keepAlivePeriod, dialTimeout time.Duration) ContextDialer {
```

Comment:
How about exporting `NewDirectDialer` and `NewProxyDialer`?
`NewDialer` becomes `NewDirectDialer`.
`NewTunnelDialerWithAddressDiscovery` becomes `NewProxyDialer`.
`NewTunnelDialer` is unexported because users wouldn't ever need to use it (but correct me if I'm wrong)